### ISSUE TITLE
[Feature Request] Start armed option for multiplayer

### ISSUE THREAD

**Issue by Graslu** [fgsfdsfgs/perfect_dark#144]
I may be mixing it up with GoldenEye, but I think I remember a mod that made it possible to make everyone start with the slot 1 weapon in multiplayer. I think this can greatly improve the multiplayer experience, but I understand it's also low priority to add options like this.

**Comment by LanHikariDS** [comment]
The TimeSplitters games also featured such an option, which you may be thinking of as well.

**Comment by Phnod** [comment]
I like this idea, wanted to implement it myself for fun earlier since I think it'd be easy but wasn't sure how the text label in the menu would be handled, is there a way to add custom labels currently or would that require a separate feature? Said feature would be extremely handy in the future for toggling new and previously added enhancements without having to go through the ini everytime.

**Comment by fgsfdsfgs** [comment]
There is not. Well, technically you can pass a function instead of a string index, but that's a bit awkward.
I still don't know what would be the best way to implement this.
Adding an extra string bank kinda works and is probably the best way to do it, because we could leverage the existing conversion system, but that would require me to either include it in the executable or bundle it as an external data file.
A way to write string literals in place of string table indices might be simpler and easier, but that might prove a problem later if we decide to add multilanguage support.

**Comment by jonaeru** [comment]
I think I may have seen this mod on maxbots.
https://github.com/n64decomp/perfect_dark/commit/3681c7d75fb1127a0a26dfe5ddfbe67fa972687a

**Comment by Graslu** [comment]
>I think I may have seen this mod on maxbots.
n64decomp@3681c7d

Yes, that's it!

**Comment by rafccq** [comment]
Another interesting feature imo would be a temporary invincibility on spawn - lets say 5s (maybe configurable) - which ends if you grab a weapon. There are so many possibilities for enhancements in the future.

**Comment by Graslu** [comment]
Custom invincibility timer would be interesting. 3 second by default with it turning off when shooting (start armed) or picking up a weapon would be nice for packed multiplayer matches if the bot increase / online does happen down the line.

**Comment by 2023PerfectDark** [comment]
Yea a spawn protection option would be acceptable, especially for smaller maps and increased player counts. I just know it will be annoying on the receiving end with harder Sim challenges if they respawn near you.

**Comment by LanHikariDS** [comment]
> Yea a spawn protection option would be acceptable, especially for smaller maps and increased player counts. I just know it will be annoying on the receiving end with harder Sim challenges if they respawn near you.

Well, in practice it shouldn't be enabled for Challenges, so this shouldn't be an issue

**Comment by AXDOOMER** [comment]
> I think I remember a mod that made it possible to make everyone start with the slot 1 weapon in multiplayer

That's my mod. I implemented this in GoldenEye. Works really great and I think that'd be really helpful in PD because guns are often already taken when a player respawns in 8 player games, leaving players harmless. 

An alternative I saw in some games was to make weapons stay on the map, but a player can't pickup the same weapon twice. Because the ammo pickups may not be present, pickuping a new weapons gives double ammo rather than the normal amount.
